### PR TITLE
[Ability] Implement Gorilla Tactics

### PR DESCRIPTION
**Pull request by torranx**
## What are the changes the user will see?

[Gorilla Tactics](https://bulbapedia.bulbagarden.net/wiki/Gorilla_Tactics_(Ability)) implemented

## Why am I making these changes?

Ability is unimplemented

## What are the changes from a developer perspective?

`GorillaTacticsAbAttr` - ability's class attr
`GorillaTacticsTag` - battler tag for the ability

Updates on `MoveRestrictionBattlerTag`
- made the `interruptedText` method optional

## How to test the changes?
use Gorilla Tactics ability -> play the game
or
`npm run test gorilla_tactics`

## Checklist
- [x] **I'm using `beta` as my base branch**
- [x] There is no overlap with another PR?
- [x] The PR is self-contained and cannot be split into smaller PRs?
- [x] Have I provided a clear explanation of the changes?
- [x] Have I considered writing automated tests for the issue?
- [x] If I have text, did I make it translatable and add a key in the English locale file(s)?
- [x] Have I tested the changes (manually)?
    - [x] Are all unit tests still passing? (`npm run test`)
- [ ] Are the changes visual?
  - [ ] Have I provided screenshots/videos of the changes?

### CODE REVIEW

File: src/data/battler-tags.ts [hunk 1]
```diff
@@ -207,8 +211,12 @@ export class DisabledTag extends MoveRestrictionBattlerTag {
...
   }
 
-  /** @override */
-  override interruptedText(pokemon: Pokemon, move: Moves): string {
+  /**
+   * @param {Pokemon} pokemon {@linkcode Pokemon} attempting to use the restricted move
+   * @param {Moves} move {@linkcode Moves} ID of the move being interrupted
+   * @returns {string} text to display when the move is interrupted
+   */
+  interruptedText(pokemon: Pokemon, move: Moves): string {
```
Comment (DayKev) [inline src/data/battler-tags.ts:219]: Why did you remove `override`?

File: src/data/battler-tags.ts [hunk 2]
```diff
@@ -219,6 +227,72 @@ export class DisabledTag extends MoveRestrictionBattlerTag {
...
+   * @param {Pokemon} pokemon the {@linkcode Pokemon} to check if the tag can be added
+   * @returns `true` if the pokemon has a valid move and no existing {@linkcode GorillaTacticsTag}; `false` otherwise
+   */
+  override canAdd(pokemon: Pokemon): boolean {
+    return (this.getLastValidMove(pokemon) !== undefined) && !pokemon.getTag(GorillaTacticsTag);
+  }
+
+  /**
+   * Ensures that move history exists on {@linkcode Pokemon} and has a valid move.
+   * If so, sets the {@linkcode moveId} and increases the user's Attack by one stage.
```
Comment (DayKev) [inline src/data/battler-tags.ts:257]:
```suggestion
   * If so, sets the {@linkcode moveId} and increases the user's Attack by 50%.
```
GT doesn't alter stat stages.

Comment (torranx) [reply src/data/battler-tags.ts]: missed to update the docs

File: src/data/battler-tags.ts (hunk 1 above)
Comment (torranx) [reply src/data/battler-tags.ts]: missed to re-add when updating the base class

**Comment by DayKev** [comment]
I don't think the ability flyout should activate for GT, but on the first turn after I used a move it popped out (not in subsequent turns though).

**Review by DayKev — APPROVED** [review]

**Review by DayKev — DISMISSED** [review]

**Review by Tempo-anon — APPROVED** [review]
Time to test in beta
